diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/files/FileStructureRule.kt

```diff
@@ -188,22 +206,80 @@ class FileStructureRule(private val configRules: List<RulesConfig>) : Rule("file
                 }
             }
             .forEach { FILE_WILDCARD_IMPORTS.warn(configRules, emitWarn, isFixMode, it.text, it.startOffset, it) }
-
         val sortedImportsGroups = if (importsGroupingConfig.useRecommendedImportsOrder) {
             regroupImports(imports.map { it.psi as KtImportDirective })
                 .map { group -> group.map { it.node } }
         } else {
             listOf(imports)
         }
             .map { group -> group.sortedBy { it.text } }
-
         if (sortedImportsGroups.flatten() != imports) {
             FILE_UNORDERED_IMPORTS.warnAndFix(configRules, emitWarn, isFixMode, "${sortedImportsGroups.flatten().first().text}...", node.startOffset, node) {
                 rearrangeImports(node, imports, sortedImportsGroups)
             }
         }
     }
 
+    @Suppress("UnsafeCallOnNullableType")
+    private fun checkUnusedImport(
+        node: ASTNode,
+        unusedImportConfig: UnusedImportConfig
+    ) {
+        findAllReferences(node)
+        packageName = (node.findChildByType(PACKAGE_DIRECTIVE)?.psi as KtPackageDirective).qualifiedName
+        node.findChildByType(IMPORT_LIST)?.getChildren(TokenSet.create(IMPORT_DIRECTIVE))?.toList()
+            ?.forEach { import ->
+                val ktImportDirective = import.psi as KtImportDirective
+                val importName = ktImportDirective.importPath?.importedName?.asString()
+                val importPath = ktImportDirective.importPath?.pathStr!!  // importPath - ifNOtParsed & Nullable
+                if (ktImportDirective.aliasName == null &&
+                        packageName.isNotEmpty() && importPath.startsWith("$packageName.") &&
+                        importPath.substring(packageName.length + 1).indexOf('.') == -1
+                ) {
+                    if (unusedImportConfig.deleteUnusedImport) {
```

Comment:
```suggestion
                    // this branch corresponds to imports from the same package
                    if (unusedImportConfig.deleteUnusedImport) {
```
We generally try to add comments in code, when it's not obvious. Here 4 conjugated conditions seem unobvious enough.